worker/uniter/storage/resolver.go

```diff
@@ -70,14 +65,24 @@ func (s *storageResolver) NextOp(
 		return nil, errors.Trace(err)
 	}
 
+	// Only IAAS models need to first run the install hook.
+	// For CAAS models, storage is specified in the pod config
+	// and mounted as the pod in started.
+	blockedWaitingForInstall := !localState.Installed && s.modelType == model.IAAS
```

Comment:
Changing this to !localState.Installed, didn't work, CAAS units with storage hand.